machineconfiguration/v1alpha1/types_osimagestream.go

```diff
@@ -0,0 +1,126 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=osimagestreams,scope=Cluster
+// +kubebuilder:subresource:status
+// +openshift:api-approved.openshift.io=https://github.com/openshift/api/pull/2555
+// +openshift:file-pattern=cvoRunLevel=0000_80,operatorName=machine-config,operatorOrdering=01
+// +openshift:enable:FeatureGate=OSStreams
+// +kubebuilder:metadata:labels=openshift.io/operator-managed=
+
+// OSImageStream describes a set of streams and associated URLs available
+// for the MachineConfigPools to be used as base OS images.
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+type OSImageStream struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec contains the desired OSImageStream config configuration.
+	// +required
+	Spec OSImageStreamSpec `json:"spec"`
+
+	// status describes the last observed state of this OSImageStream.
+	// Populated by the MachineConfigOperator after reading release metadata.
+	// When not present, the controller has not yet reconciled this resource.
+	// +optional
+	Status OSImageStreamStatus `json:"status"`
+}
+
+// OSImageStreamStatus describes the current state of a OSImageStream.
+// +kubebuilder:validation:XValidation:rule="size(self.availableStreams) == 0 || self.defaultStream != ''",message="defaultStream must be set when availableStreams is not empty"
+// +kubebuilder:validation:XValidation:rule="!has(self.defaultStream) || self.defaultStream in self.availableStreams.map(s, s.name)",message="defaultStream must reference a stream name from availableStreams"
+type OSImageStreamStatus struct {
+	// conditions represents the latest available observations of current state.
+	// When empty, no conditions have been reported yet.
+	// +listType=map
+	// +listMapKey=type
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
```

Comment:
Curious what conditions would be reflected here? This object would mostly be there to list available streams from the payload. It shouldn't really "fail" or "progress" since no user action is required.